ava-labs/core-mobile
# CP-8437 Sign messages

## Description

**Ticket: [CP-8437]** 

This PR continues on @ruijialin-avalabs work and enables signing messages on core.app with `avalanche_signMessage` rpc method.
Portion of this fix is [fixing core.app](https://github.com/ava-labs/core-web-properties/pull/2771) to not iterate through 10000 addresses if wallet (seedless) doesn't have that many.


## Screenshots/Videos
### Signing with Seedless wallet

https://github.com/ava-labs/avalanche-wallet-apps/assets/6573904/123c86eb-8a13-4d8b-bb32-a498620e925e

### Signing with Mnemonic wallet

https://github.com/ava-labs/avalanche-wallet-apps/assets/6573904/71fce756-8774-4231-b2d2-95bbd2ad44e1


## Testing
1. connect your HD wallet in Core mobile
2. connect mobile wallet with core.app
3. go to https://core.app/tools/signing-tools/sign/ 
4. select Avalance P (or X) from dropdown
5. select address
6. type in message
7. press `Sign Message`
8. confirm signing on mobile
9. verify message 

Repeat steps 2 - 9 but with Seedless wallet

## Checklist

Please check all that apply (if applicable)


[CP-8437]: https://ava-labs.atlassian.net/browse/CP-8437?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
